defold-spine/commonsrc/vertices.cpp

```diff
@@ -35,6 +37,83 @@ static uint32_t EnsureArrayFitsNumber(dmArray<T>& array, uint32_t num_to_add)
+    bounds.minX = FLT_MAX;
+    bounds.minY = FLT_MAX;
+    bounds.maxX = -FLT_MAX;
+    bounds.maxY = -FLT_MAX;
+
+    // For each slot in the draw order array of the skeleton
+    for (int s = 0; s < skeleton->slotsCount; ++s)
+    {
+        spSlot* slot = skeleton->drawOrder[s];
+
+        // Fetch the currently active attachment, continue
+        // with the next slot in the draw order if no
+        // attachment is active on the slot
+        spAttachment* attachment = slot->attachment;
+        if (!attachment)
+        {
+            continue;
+        }
+
+        // Fill the vertices array depending on the type of attachment
+        if (attachment->type == SP_ATTACHMENT_REGION)
+        {
+            // Cast to an spRegionAttachment so we can get the rendererObject
+            // and compute the world vertices
+            spRegionAttachment* regionAttachment = (spRegionAttachment*)attachment;
+
+            // Computed the world vertices positions for the 4 vertices that make up
+            // the rectangular region attachment. This assumes the world transform of the
+            // bone to which the slot (and hence attachment) is attached has been calculated
+            // before rendering via spSkeleton_updateWorldTransform
+            spRegionAttachment_computeWorldVertices(regionAttachment, slot->bone, scratch.Begin(), 0, 2);
+        }
+        else if (attachment->type == SP_ATTACHMENT_MESH)
+        {
+            // Cast to an spMeshAttachment so we can get the rendererObject
+            // and compute the world vertices
+            spMeshAttachment* mesh = (spMeshAttachment*)attachment;
+
+            int num_world_vertices = mesh->super.worldVerticesLength / 2;
+
+            EnsureArrayFitsNumber(scratch, num_world_vertices*2); // increase capacity if needed
+
+            // Computed the world vertices positions for the vertices that make up
+            // the mesh attachment. This assumes the world transform of the
+            // bone to which the slot (and hence attachment) is attached has been calculated
+            // before rendering via spSkeleton_updateWorldTransform
+
+            spVertexAttachment_computeWorldVertices(SUPER(mesh), slot, 0, num_world_vertices*2, scratch.Begin(), 0, 2);
+        }
+
+        if (attachment->type == SP_ATTACHMENT_REGION || attachment->type == SP_ATTACHMENT_MESH) {
+            // go through vertex coords and update max/min for X and Y
+            float* coords = scratch.Begin();
+            uint32_t vertex_count = scratch.Size()/2; // since stride is 2
+
+            for (int i=0; i<vertex_count; i++) {
+                float x = *coords++;
+                float y = *coords++;
+                bounds.minX = dmMath::Min(x, bounds.minX);
+                bounds.minY = dmMath::Min(y, bounds.minY);
+                bounds.maxX = dmMath::Max(x, bounds.maxX);
+                bounds.maxY = dmMath::Max(y, bounds.maxY);
+            }
+            scratch.SetSize(0);
```

Comment:
Not sure why you need to set the size here?
The idea of the scratch buffer is that it may grow, and should always fit the data.

Since we don't set the size again for the `SP_ATTACHMENT_REGION`, we will now write to a dmArray which has size 0 which I honestly feel is misleading and error prone.

I recommend just removing the "SetSize(0)".

Reply:
ok.

But i'll have to keep somewhere the number of vertices involved. I'll initialize `vertex_count` eariler then.